src/Conan.cmake

```diff
@@ -217,7 +217,11 @@ macro(_run_conan2)
   endif()
 
   if(NOT _args_INSTALL_ARGS)
-    set(_args_INSTALL_ARGS "--build=missing")
+    set(_args_INSTALL_ARGS
+        --build=missing
+        -c tools.system.package_manager:mode=install
+        -c tools.system.package_manager:sudo=True
+    )
```

Comment:
> > Does this work with Conan 1 as well as 2?
> 
> Only conan 2. Currently [cmake-conan integration for Conan 1.X is no longer mantainedand](https://github.com/conan-io/cmake-conan/tree/develop?rgh-link-date=2024-11-10T06%3A18%3A18Z) and [Conan Center has stopped receiving updates for Conan 1.x packages](https://blog.conan.io/2024/09/30/Conan-Center-will-stop-receiving-updates-for-Conan-1.html), so I didn't consider how to enable it for conan 1.

Could you conditionally add the flags for Conan 2?

Reply:
It's not necessary. The function `_run_conan2` here is just for conan 2. The user interface `run_conan` checks user's conan version and dispatch the call to either `_run_conan2` or `_run_conan1`